### ISSUE TITLE
ErrorException Undefined index: user

### ISSUE THREAD

**Issue by Grummfy** [owen-it/laravel-auditing#282]
| Q                 | A
| ----------------- | ---
| Bug?              | yes
| New Feature?      | no
| Framework         | Laravel (|Lumen?)
| Framework version | 5.4.30
| Package version   | 4.1.1
| PHP version       | 7.1.6

#### Actual Behaviour

An error of undefined index appears

#### Expected Behaviour

No error ;)

#### Steps to Reproduce

```php
dump(MyModel::find($id)->audits()->first()->getMetadata());
```
_ErrorException Undefined index: user_

no error with:

```php
dump(MyModel::find($id)->audits()->with('user')->first()->getMetadata());
```

#### Possible Solutions

inside \OwenIt\Auditing\Audit::resolveData
replace
```php
if ($user = $this->getRelation('user')) {
```
by
```php
if ($user = $this->user) {
```

**Comment by quetzyg** [comment]
Show me the `User` class, the `MyModel` class and the `audit.php` configuration file you have.

**Comment by Grummfy** [comment]
audit.php

```php
<?php

return [

    /*
    |--------------------------------------------------------------------------
    | Audit implementation
    |--------------------------------------------------------------------------
    |
    | Define which Audit model implementation should be used.
    |
    */

    'implementation' => OwenIt\Auditing\Models\Audit::class,

    /*
    |--------------------------------------------------------------------------
    | User Keys, Model & Resolver
    |--------------------------------------------------------------------------
    |
    | Define the User primary and foreign keys, Eloquent model and ID resolver
    | class.
    |
    */

    'user' => [
        'primary_key' => 'id',
        'foreign_key' => 'user_id',
        'model'       => \App\Models\Database\User::class,
        'resolver'    => function()
        {
            return Auth::check() ? Auth::user()->getAuthIdentifier() : null;
        },
    ],

    /*
    |--------------------------------------------------------------------------
    | Default Driver
    |--------------------------------------------------------------------------
    |
    | The default audit driver used to keep track of changes.
    |
    */

    'default' => 'database',

    /*
    |--------------------------------------------------------------------------
    | Audit Drivers
    |--------------------------------------------------------------------------
    |
    | Available audit drivers and respective configurations.
    |
    */
    'drivers' => [
        'database' => [
            'table'      => 'audits',
            'connection' => env('DB_CONNECTION', 'mysql'),
        ],
    ],

    /*
    |--------------------------------------------------------------------------
    | Audit Console?
    |--------------------------------------------------------------------------
    |
    | Whether we should audit console events (eg. php artisan db:seed).
    |
    */

    'console' => false,
];
```

MyModel is just one model around several others, it's similar to this one
```php
<?php
namespace App\Models\Database;

use App\Models\Model as Eloquent;
use OwenIt\Auditing\Auditable;
use OwenIt\Auditing\Contracts\Auditable as AuditableContract;

class MyModel extends Eloquent implements AuditableContract
{
	use Auditable;
}
```

User
```php
<?php

namespace App\Models\Database;

use App\Models\Model as Eloquent;
use OwenIt\Auditing\Auditable;
use OwenIt\Auditing\Contracts\Auditable as AuditableContract;

class User extends Eloquent implements AuditableContract
{
	use Auditable;
}
```

I try to use model as basic as possible and I got the same behaviour. 

Note : all my dev are made with error_reporting = E_ALL

**Comment by quetzyg** [comment]
Sorry, I can't reproduce the issue. You say that if you replace
```php
if ($user = $this->getRelation('user')) {
```
 with 
```php
if ($user = $this->user) {
```
it works?

If you `dd($this->user)` do you get a `null` or a `User` instance?

**Comment by Grummfy** [comment]
I got a user or null following if there is a connected user.
But the fact that it's null or not, change nothing on the error.

**Comment by quetzyg** [comment]
>I got a user or null following if there is a connected user.
Can you rephrase that?

>But the fact that it's null or not, change nothing on the error.
If you have `null`, it might explain why you get that issue.

**Comment by Grummfy** [comment]
I try to reproduce the case with a clean fresh laravel installation with just this package (so with the default user model).

Foo.php
```php
<?php

namespace App;

use Illuminate\Database\Eloquent\Model;

use OwenIt\Auditing\Auditable;
use OwenIt\Auditing\Contracts\Auditable as AuditableContract;

class Foo extends Model  implements AuditableContract
{

	use Auditable;
    //
public $timestamps = false;
	protected $fillable = [
	'name',
];
}
```

Simple table, with a simple attribute
```php
 \App\Foo::create(['name' => 'abc-' . uniqid()]);
```

```php
	foreach(\App\Foo::all() as $foo)
	{
		dump($foo->audits()->first()->getMetadata());
	}

```
exactly the same error.

**Comment by quetzyg** [comment]
Try: `dump($foo->audits()->with('user')->first()->getMetadata());`

**Comment by Grummfy** [comment]
With ```->with('user')``` it's working perfectly. whatever the value of ```->user``` is.
But, I'm not sure that we require to do that each time we want to access to metadata.

**Comment by quetzyg** [comment]
Yeah, I'll have to change that again. We shouldn't be forced to pass `->with('user')`.

**Comment by quetzyg** [comment]
Fixed in the latest version.

**Comment by Grummfy** [comment]
Thanks. That was fast
